server/taskflows/hpccloud/taskflow/nwchem.py

```diff
@@ -0,0 +1,290 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#  Copyright 2016 Kitware Inc.
+#
+#  Licensed under the Apache License, Version 2.0 ( the "License" );
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+###############################################################################
+import tempfile
+import json
+import os
+import subprocess
+import shutil
+from ConfigParser import SafeConfigParser
+from jsonpath_rw import parse
+from bson.objectid import ObjectId
+
+import cumulus.taskflow
+from cumulus.tasks.job import download_job_input_folders, submit_job
+from cumulus.tasks.job import monitor_job, monitor_jobs
+from cumulus.tasks.job import upload_job_output_to_folder
+from cumulus.tasks.job import terminate_job
+from cumulus.transport.files.download import download_path_from_cluster
+from girder.utility.model_importer import ModelImporter
+from girder.api.rest import getCurrentUser
+from girder.constants import AccessType
+from girder_client import GirderClient, HttpError
+
+from hpccloud.taskflow.utility import *
+
+class NwChemTaskFlow(cumulus.taskflow.TaskFlow):
+    """
+    {
+        "input": {
+            "folder": {
+                "id": <the id of the folder containing input files>
+            },
+            "geometry": {
+                "id": <the file id of the geometry file>
+            }
+        },
+        "output": {
+            "folder": {
+                "id": <id of folder to upload output into>
+            }
+        },
+        "cluster": {
+            "_id": <id of cluster to run on>
+        }
+    }
+    """
+    NWCHEM_IMAGE = {
+        'name': 'NWChem_ParaView-5.0.1',
+        'owner': '695977956746'
+    }
+
+    def start(self, *args, **kwargs):
+        user = getCurrentUser()
+        # Load the cluster
+        # TODO: should this be in a common class?
```

Comment:
:+1: Definitely if we can factor this out from the three taskflow we have that would be great.